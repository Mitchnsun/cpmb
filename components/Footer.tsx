import Image from "next/image";
import Link from "next/link";

import artists from "@/assets/contents/artists.json";
import Heading from "@/components/Heading";

const Footer = () => {
  const artistNames = Object.keys(artists);

  return (
    <footer className="font-noto">
      <section className="container mx-auto flex items-center justify-around gap-6 py-4 text-center">
        <a
          href="https://www.veran-piano.com/"
          aria-label="Visitez le site de Veran Piano - nouvelle fenêtre"
          target="_blank"
          rel="noopener noreferrer"
          className="basis-1/3"
        >
          <Image
            src="/logo-veran-pianos.png"
            alt="Logo Veran Piano"
            width={200}
            height={120}
            sizes="200px"
            className="m-auto h-auto"
          />
        </a>
        <a
          href="https://www.gaillard.fr/"
          aria-label="Visitez le site de la Ville de Gaillard - nouvelle fenêtre"
          target="_blank"
          rel="noopener noreferrer"
          className="basis-1/3"
        >
          <Image
            src="/logo-gaillard.png"
            alt="Logo Ville de Gaillard"
            width={200}
            height={120}
            sizes="200px"
            className="m-auto h-auto"
          />
        </a>
        <a
          href="https://www.hautesavoie.fr/"
          aria-label="Visitez le site du Département de la Haute-Savoie - nouvelle fenêtre"
          target="_blank"
          rel="noopener noreferrer"
          className="basis-1/3"
        >
          <Image
            src="/haute-savoie.svg"
            alt="Logo Haute-Savoie"
            width={90}
            height={55}
            sizes="90px"
            className="m-auto h-auto"
          />
        </a>
      </section>
      <div className="bg-gray-800 px-4">
        <div className="container mx-auto flex flex-col items-start gap-4 py-4 lg:flex-row lg:gap-24">
          <div>
            <Image
              src="/CPMB-logo-blanc.png"
              alt="Logo Chœur des Pays du Mont-Blanc"
              className="h-auto shrink-0"
              sizes="160px"
              width={160}
              height={55}
            />
            <p className="mt-4 text-sm text-gray-300">Partager la passion de la musique chorale au cœur des Alpes.</p>
          </div>
          <section>
            <Heading hLevel={4} variant={3} className="mb-2 text-white">
              Nos Partenaires
            </Heading>
            <ul className="space-y-1 text-sm text-gray-300">
              <li>
                <a
                  href="https://www.veran-piano.com/"
                  aria-label="Visitez le site de Veran Piano - nouvelle fenêtre"
                  className="underline-offset-2 hover:text-white focus-visible:underline"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Veran Piano
                </a>
              </li>
              <li>
                <a
                  href="https://www.gaillard.fr/"
                  aria-label="Visitez le site de la Ville de Gaillard - nouvelle fenêtre"
                  className="underline-offset-2 hover:text-white focus-visible:underline"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Ville de Gaillard
                </a>
              </li>
              <li>
                <a
                  href="https://www.hautesavoie.fr/"
                  aria-label="Visitez le site du Département de la Haute-Savoie - nouvelle fenêtre"
                  className="underline-offset-2 hover:text-white focus-visible:underline"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Département de la Haute-Savoie
                </a>
              </li>
            </ul>
          </section>
          <section>
            <Heading hLevel={4} variant={3} className="mb-2 text-white">
              Plan du site
            </Heading>
            <ul className="space-y-1 text-sm text-gray-300">
              <li>
                <Link href="/" className="underline-offset-2 hover:text-white focus-visible:underline">
                  Page d&apos;accueil
                </Link>
              </li>
              <li>
                <Link href="/presentation" className="underline-offset-2 hover:text-white focus-visible:underline">
                  Présentation
                </Link>
              </li>
              <li>
                <Link href="/nos-concerts" className="underline-offset-2 hover:text-white focus-visible:underline">
                  Nos concerts
                </Link>
              </li>
              <li>
                <Link href="/presse" className="underline-offset-2 hover:text-white focus-visible:underline">
                  Presse
                </Link>
              </li>
              <li>
                <Link href="/mentions-legales" className="underline-offset-2 hover:text-white focus-visible:underline">
                  Mentions légales
                </Link>
              </li>
            </ul>
          </section>
          <section>
            <Heading hLevel={4} variant={3} className="mb-2 text-white">
              Artistes
            </Heading>
            <ul className="space-y-1 text-sm text-gray-300">
              {artistNames.map((artistKey) => (
                <li key={artistKey}>
                  <Link
                    href={`/presentation/${artistKey}`}
                    className="underline-offset-2 hover:text-white focus-visible:underline"
                  >
                    {artists[artistKey as keyof typeof artists].name}
                  </Link>
                </li>
              ))}
            </ul>
          </section>
        </div>
        <p className="container mx-auto border-t border-gray-700 py-4 text-center text-sm text-gray-300">
          &copy; {new Date().getFullYear()} Chœur des Pays du Mont-Blanc. Tous droits réservés - Conçu et développé
          par&nbsp;
          <a
            href="https://www.gocosmic.dev/fr"
            target="_blank"
            rel="noopener noreferrer"
            aria-label="Visitez le site de Go Cosmic - nouvelle fenêtre"
            className="text-sky-400 underline decoration-sky-400 underline-offset-2 hover:text-sky-300 focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-gray-800 focus-visible:outline-none"
          >
            Go Cosmic
          </a>
        </p>
      </div>
    </footer>
  );
};

export default Footer;
