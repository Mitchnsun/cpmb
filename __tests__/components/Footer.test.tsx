import { render } from "@testing-library/react";

import Footer from "@/components/Footer";

describe("Footer", () => {
  beforeEach(() => {
    // Mock Date.getFullYear() to return a consistent year for testing
    vi.useFakeTimers();
    vi.setSystemTime(new Date("2024-01-01"));
  });

  afterEach(() => {
    vi.useRealTimers();
  });

  it("should render the footer element", () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
