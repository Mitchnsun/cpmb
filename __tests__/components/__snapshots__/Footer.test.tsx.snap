// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should render the footer element 1`] = `
<div>
  <footer
    class="font-noto"
  >
    <section
      class="container mx-auto flex items-center justify-around gap-6 py-4 text-center"
    >
      <a
        aria-label="Visitez le site de Veran Piano - nouvelle fenêtre"
        class="basis-1/3"
        href="https://www.veran-piano.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Logo Veran Piano"
          class="m-auto h-auto"
          height="120"
          src="/logo-veran-pianos.png"
          width="200"
        />
      </a>
      <a
        aria-label="Visitez le site de la Ville de Gaillard - nouvelle fenêtre"
        class="basis-1/3"
        href="https://www.gaillard.fr/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Logo Ville de Gaillard"
          class="m-auto h-auto"
          height="120"
          src="/logo-gaillard.png"
          width="200"
        />
      </a>
      <a
        aria-label="Visitez le site du Département de la Haute-Savoie - nouvelle fenêtre"
        class="basis-1/3"
        href="https://www.hautesavoie.fr/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Logo Haute-Savoie"
          class="m-auto h-auto"
          height="55"
          src="/haute-savoie.svg"
          width="90"
        />
      </a>
    </section>
    <div
      class="bg-gray-800 px-4"
    >
      <div
        class="container mx-auto flex flex-col items-start gap-4 py-4 lg:flex-row lg:gap-24"
      >
        <div>
          <img
            alt="Logo Chœur des Pays du Mont-Blanc"
            class="h-auto shrink-0"
            height="55"
            src="/CPMB-logo-blanc.png"
            width="160"
          />
          <p
            class="mt-4 text-sm text-gray-300"
          >
            Partager la passion de la musique chorale au cœur des Alpes.
          </p>
        </div>
        <section>
          <h4
            class="font-bold tracking-tight text-lg mb-2 text-white"
          >
            Nos Partenaires
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-300"
          >
            <li>
              <a
                aria-label="Visitez le site de Veran Piano - nouvelle fenêtre"
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="https://www.veran-piano.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Veran Piano
              </a>
            </li>
            <li>
              <a
                aria-label="Visitez le site de la Ville de Gaillard - nouvelle fenêtre"
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="https://www.gaillard.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Ville de Gaillard
              </a>
            </li>
            <li>
              <a
                aria-label="Visitez le site du Département de la Haute-Savoie - nouvelle fenêtre"
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="https://www.hautesavoie.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Département de la Haute-Savoie
              </a>
            </li>
          </ul>
        </section>
        <section>
          <h4
            class="font-bold tracking-tight text-lg mb-2 text-white"
          >
            Plan du site
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-300"
          >
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/"
              >
                Page d'accueil
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/presentation"
              >
                Présentation
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/nos-concerts"
              >
                Nos concerts
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/presse"
              >
                Presse
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/mentions-legales"
              >
                Mentions légales
              </a>
            </li>
          </ul>
        </section>
        <section>
          <h4
            class="font-bold tracking-tight text-lg mb-2 text-white"
          >
            Artistes
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-300"
          >
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/presentation/benoit-dubu"
              >
                Benoît Dubu
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/presentation/agnes-lorincz"
              >
                Ágnes Lőrincz
              </a>
            </li>
            <li>
              <a
                class="underline-offset-2 hover:text-white focus-visible:underline"
                href="/presentation/marjorie-saunier"
              >
                Marjorie Saunier
              </a>
            </li>
          </ul>
        </section>
      </div>
      <p
        class="container mx-auto border-t border-gray-700 py-4 text-center text-sm text-gray-300"
      >
        © 
        2024
         Chœur des Pays du Mont-Blanc. Tous droits réservés - Conçu et développé par 
        <a
          aria-label="Visitez le site de Go Cosmic - nouvelle fenêtre"
          class="text-sky-400 underline decoration-sky-400 underline-offset-2 hover:text-sky-300 focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-gray-800 focus-visible:outline-none"
          href="https://www.gocosmic.dev/fr"
          rel="noopener noreferrer"
          target="_blank"
        >
          Go Cosmic
        </a>
      </p>
    </div>
  </footer>
</div>
`;
